lib/src/booleans/boolean_disjoint.dart

```diff
@@ -0,0 +1,365 @@
+import '../../helpers.dart';
+import '../../meta.dart';
+import '../line_intersect.dart';
+import '../polygon_to_line.dart';
+import 'boolean_point_in_polygon.dart';
+
+/**
+ * Boolean-disjoint returns (TRUE) if the intersection of the two geometries is an empty set.
+ *
+ * @name booleanDisjoint
+ * @param {Geometry|Feature<any>} feature1 GeoJSON Feature or Geometry
+ * @param {Geometry|Feature<any>} feature2 GeoJSON Feature or Geometry
+ * @returns {boolean} true/false
+ * @example
+ * var point = turf.point([2, 2]);
+ * var line = turf.lineString([[1, 1], [1, 2], [1, 3], [1, 4]]);
+ *
+ * turf.booleanDisjoint(line, point);
+ * //=true
+ */
+bool booleanDisjoint(GeoJSONObject feature1, GeoJSONObject feature2) {
+  var bool = true;
+  flattenEach(feature1, (flatten1, featureIndex, multiFeatureIndex) {
+    flattenEach(feature2, (flatten2, featureIndex, multiFeatureIndex) {
+      if (bool == false) {
+        return false;
+      }
+      bool = disjoint(flatten1.geometry!, flatten2.geometry!);
+    });
+  });
+  return bool;
+}
+
+/**
+ * Disjoint operation for simple Geometries (Point/LineString/Polygon)
+ *
+ * @private
+ * @param {Geometry<any>} geom1 GeoJSON Geometry
+ * @param {Geometry<any>} geom2 GeoJSON Geometry
+ * @returns {boolean} true/false
+ */
+disjoint(GeometryType geom1, GeometryType geom2) {
+  switch (geom1.type) {
+    case GeoJSONObjectType.point:
+      switch (geom2.type) {
+        case GeoJSONObjectType.point:
+          return !compareCoords(geom1.coordinates, geom2.coordinates);
+        case GeoJSONObjectType.lineString:
+          return !isPointOnLine(geom2 as LineString, geom1 as Point);
+        case GeoJSONObjectType.polygon:
+          return !booleanPointInPolygon((geom1 as Point).coordinates, geom2);
+      }
+      /* istanbul ignore next */
+      break;
+    case GeoJSONObjectType.lineString:
+      switch (geom2.type) {
+        case GeoJSONObjectType.point:
+          return !isPointOnLine(geom1 as LineString, geom2 as Point);
+        case GeoJSONObjectType.lineString:
+          return !isLineOnLine(geom1 as LineString, geom2 as LineString);
+        case GeoJSONObjectType.polygon:
+          return !isLineInPoly(geom2 as Polygon, geom1 as LineString);
+      }
+      /* istanbul ignore next */
+      break;
```

Comment:
please check with `if(type is Point)` instead of the switch case, we can remove a lot of explicit casting here